### PR TITLE
Setup api

### PR DESCRIPTION
**Pull request by josuerushanika**
In this activity I'm  working on the Leaderboard list app, sending and receiving data from/to the Leaderboard API:

_Project requirements_

Read the [Leaderboard API documentation](https://www.notion.so/microverse/Leaderboard-API-service-24c0c3c116974ac49488d4eb0267ade3) to learn how to use this API.
Create a new game with the name of your choice by using the API.
Make sure that you saved the ID of your game that will be returned by API.
Implement the "Refresh" button that will get all scores for a game created by you from the API (receiving data from the API and parsing the JSON).
Implement the form "Submit" button that will save a score for a game created by you (sending data to the API).
Use arrow functions instead of the function keyword.
Use async and await JavaScript features to consume the API.
No styling is required.

### CODE REVIEW

File: src/modules/MethodsAPI.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+    body: JSON.stringify({ user: newPlayer.user, score: newPlayer.score }),
+  });
+  console.log('Getpost', post);
+
+  const response = await post.json();
+  return response;
+};
```
Comment (anagudelogu) [inline src/modules/MethodsAPI.js]: - [ ] Please consider using the correct naming convention for this variable, at the moment, you're using `pascal case` which is most widely used for classes. The naming convention for variables and functions should be in `camel case`, so please apply this change! 🤓 🔧 

- [ ] Also, according to the [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md), please avoid committing console.log calls: 🤓 🔧 

image

File: src/modules/MethodsAPI.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  try {
+    const get = await fetch(baseUrl);
+    const response = await get.json();
+    console.log('myResponse', response.result);
+    const sortedGameData = response.result.sort((a, b) => b.score - a.score);
+    htmldisplay(sortedGameData);
+  } catch (err) {
+    console.log(err.message);
+  }
+};
```
Comment (anagudelogu) [inline src/modules/MethodsAPI.js]: - [ ] According to the [JavaScript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md), please avoid committing console.log calls: 🤓 🔧 

image

**Review by anagudelogu — CHANGES_REQUESTED** [review]
# STATUS:  CHANGES REQUIRED  ♻️ 🚧 

Hi @josuerushanika, 
I'm Andres.


While you have done a great job working on this milestone 💻 , there are some issues that you still need to work on to go to the next project. ♻️ 

You are almost there, keep up the good work! 🚀 



## Highlights
✅ Great use of Git flow. 💯 
✅ Your project is documented professionally. 📜 
✅ Descriptive PR title, summary and commit messages. 💬 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/modules/MethodsAPI.js (hunk 1 above)
Comment (josuerushanika) [reply src/modules/MethodsAPI.js]: It is fixed

File: src/modules/MethodsAPI.js (hunk 2 above)
Comment (josuerushanika) [reply src/modules/MethodsAPI.js]: It is fixed

**Review by AmaduKamara — APPROVED** [review]
# Status: Approved 🟢 🟢 🟢

Hi @josuerushanika,

Good job working on the changes asked by the previous reviewer 👍🏽

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! 🎉 

## Highlights:

✅ You used Gitflow correctly 👍🏽
✅ Linter checks passed 👍🏽
✅ Used arrow f=unction syntax and async-await 👍🏽

## Optional Changes ♻️

- [ ] After adding a leader and their score, when I hit the refresh button, the data is show double. Ideally, they're not supposed to repeat when the refresh button is clicked but fetch and display exactly the data from the API. Kindly check that out please 👍🏽

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**